File: SproInstall/SocketPro/tutorial/CSharp/SampleOne/Client/TOne.cs

using System;
using SocketProAdapter;
using SocketProAdapter.ClientSide;
using USOCKETLib;


public class CTOne : CAsyncServiceHandler
{
    public CTOne()
        : base(TOneConst.sidCTOne)
    {
    }

    public CTOne(CClientSocket cs)
        : base(TOneConst.sidCTOne, cs)
    {
    }
    
    public CTOne(CClientSocket cs, IAsyncResultsHandler DefaultAsyncResultsHandler)
        : base(TOneConst.sidCTOne, cs, DefaultAsyncResultsHandler)
    {
    }

	public int QueryCount()
	{
		int QueryCountRtn;
		bool bProcessRy = ProcessR1(TOneConst.idQueryCountCTOne, out QueryCountRtn);
		return QueryCountRtn;
	}

	public int QueryGlobalCount()
	{
		int QueryGlobalCountRtn;
		bool bProcessRy = ProcessR1(TOneConst.idQueryGlobalCountCTOne, out QueryGlobalCountRtn);
		return QueryGlobalCountRtn;
	}

	public int QueryGlobalFastCount()
	{
		int QueryGlobalFastCountRtn;
		bool bProcessRy = ProcessR1(TOneConst.idQueryGlobalFastCountCTOne, out QueryGlobalFastCountRtn);
		return QueryGlobalFastCountRtn;
	}

	public void Sleep(int nTime)
	{
		bool bProcessRy = ProcessR0(TOneConst.idSleepCTOne, nTime);
	}

	public object Echo(object objInput)
	{
		object EchoRtn;
		bool bProcessRy = ProcessR1(TOneConst.idEchoCTOne, objInput, out EchoRtn);
		return EchoRtn;
	}

    public void GetAllCounts(out int nCount, out int nGlobalCount, out int nGlobalFastCount)
    {
        int n = 0, global = 0, fast = 0;
        BeginBatching();
        SendRequest(TOneConst.idQueryCountCTOne, delegate(CAsyncResult ar) {
            ar.UQueue.Pop(out n);
        });
        SendRequest(TOneConst.idQueryGlobalCountCTOne, delegate(CAsyncResult ar) {
            ar.UQueue.Pop(out global);
        });
        SendRequest(TOneConst.idQueryGlobalFastCountCTOne, delegate(CAsyncResult ar) {
            ar.UQueue.Pop(out fast);
        });
        CommitBatch(true); //true -- ask server to send three results back in one batch
        WaitAll();
        nCount = n;
        nGlobalCount = global;
        nGlobalFastCount = fast;
    }

    public void GetAllCounts(out int nCount, out int nGlobalCount)
    {
        int n = 0, global = 0;
        BeginBatching();
        SendRequest(TOneConst.idQueryCountCTOne, delegate(CAsyncResult ar)
        {
            ar.UQueue.Pop(out n);
        });
        SendRequest(TOneConst.idQueryGlobalCountCTOne, delegate(CAsyncResult ar)
        {
            ar.UQueue.Pop(out global);
        });
        CommitBatch(true); //true -- ask server to send two results back in one batch
        WaitAll();
        nCount = n;
        nGlobalCount = global;
    }

    public void GetAllCounts(out int nCount)
    {
        nCount = QueryCount();
    }
}

File: SproInstall/SocketPro/tutorial/VBNet/SampleThree/Client/TOne.vb
Imports System
Imports SocketProAdapter
Imports SocketProAdapter.ClientSide
Imports USOCKETLib

Public Class CTOne
    Inherits CAsyncServiceHandler

    Public Sub New()
        MyBase.New(TOneConst.sidCTOne)
    End Sub

    Public Sub New(ByVal cs As CClientSocket)
        MyBase.New(TOneConst.sidCTOne, cs)
    End Sub

    Public Sub New(ByVal cs As CClientSocket, ByVal DefaultAsyncResultsHandler As IAsyncResultsHandler)
        MyBase.New(TOneConst.sidCTOne, cs, DefaultAsyncResultsHandler)
    End Sub

    Public Function QueryCount() As Integer
        Dim QueryCountRtn As Integer
        Dim bProcessRy As Boolean = ProcessR1(TOneConst.idQueryCountCTOne, QueryCountRtn)
        Return QueryCountRtn
    End Function

    Public Function QueryGlobalCount() As Integer
        Dim QueryGlobalCountRtn As Integer
        Dim bProcessRy As Boolean = ProcessR1(TOneConst.idQueryGlobalCountCTOne, QueryGlobalCountRtn)
        Return QueryGlobalCountRtn
    End Function

    Public Function QueryGlobalFastCount() As Integer
        Dim QueryGlobalFastCountRtn As Integer
        Dim bProcessRy As Boolean = ProcessR1(TOneConst.idQueryGlobalFastCountCTOne, QueryGlobalFastCountRtn)
        Return QueryGlobalFastCountRtn
    End Function

    Public Sub Sleep(ByVal nTime As Integer)
        Dim bProcessRy As Boolean = ProcessR0(TOneConst.idSleepCTOne, nTime)
    End Sub

    Public Function Echo(ByVal objInput As Object) As Object
        Dim EchoRtn As Object = Nothing
        Dim Groups() As Integer = {1, 3}
        'send a text message to groups 1 and 3
        GetAttachedClientSocket().Push.Broadcast("EchoData called", Groups)
        Dim bProcessRy As Boolean = ProcessR1(TOneConst.idEchoCTOne, objInput, EchoRtn)
        Return EchoRtn
    End Function

    Protected m_QueryCountRtn As Integer
    Protected Sub QueryCountAsync()
        SendRequest(TOneConst.idQueryCountCTOne)
    End Sub

    Protected m_QueryGlobalCountRtn As Integer
    Protected Sub QueryGlobalCountAsync()
        SendRequest(TOneConst.idQueryGlobalCountCTOne)
    End Sub

    Protected m_QueryGlobalFastCountRtn As Integer
    Protected Sub QueryGlobalFastCountAsync()
        SendRequest(TOneConst.idQueryGlobalFastCountCTOne)
    End Sub

    'We can process returning results inside the function.
    Protected Overrides Sub OnResultReturned(ByVal sRequestID As Short, ByVal UQueue As CUQueue)
        Select Case (sRequestID)
            Case TOneConst.idQueryCountCTOne
                UQueue.Pop(m_QueryCountRtn)
            Case TOneConst.idQueryGlobalCountCTOne
                UQueue.Pop(m_QueryGlobalCountRtn)
            Case TOneConst.idQueryGlobalFastCountCTOne
                UQueue.Pop(m_QueryGlobalFastCountRtn)
            Case Else
        End Select
    End Sub

    Public Sub GetAllCounts(ByRef nCount As Integer, ByRef nGlobalCount As Integer, ByRef nGlobalFastCount As Integer)
        BeginBatching()
        QueryCountAsync()
        QueryGlobalFastCountAsync()
        QueryGlobalCountAsync()
        CommitBatch(True) 'true -- ask server to send three results back in one batch
        WaitAll()
        nCount = m_QueryCountRtn
        nGlobalCount = m_QueryGlobalCountRtn
        nGlobalFastCount = m_QueryGlobalFastCountRtn
    End Sub

    Public Sub GetAllCounts(ByRef nCount As Integer, ByRef nGlobalCount As Integer)
        BeginBatching()
        QueryCountAsync()
        QueryGlobalCountAsync()
        CommitBatch(True) 'true -- ask server to send two results back in one batch
        WaitAll()
        nCount = m_QueryCountRtn
        nGlobalCount = m_QueryGlobalCountRtn
    End Sub

    Public Sub GetAllCounts(ByRef nCount As Integer)
        nCount = QueryCount()
    End Sub
End Class